src/collapse/dtrace.rs

```diff
@@ -136,15 +137,21 @@ impl Folder {
         }
     }
 
-    fn remove_offset(line: &str) -> &str {
-        let mut line = line.rsplitn(2, '+');
-        // at least one element will be returned
-        let second = line.next().unwrap();
-        if let Some(first) = line.next() {
-            first
-        } else {
-            second
+    fn remove_offset(line: &str) -> (bool, bool, bool, &str) {
+        let mut could_be_java = false;
+        let mut could_be_cpp = false;
+        let mut has_colon = false;
+        let bytes = line.as_bytes();
+        for offset in 0..bytes.len() {
```

Comment:
This seems really problematic to me, because it means that UTF-8 characters will break the parsing (the indexing below would crash). I think keeping Unicode support for a low-percentage performance penalty is worthwhile.

Reply:
It could be that you can use `line.chars()` instead though, and still keep your loop below.

Reply:
I'm fairly sure that dtrace ustack symbols can't be utf8.

Reply:
Is that so? I mean, maybe.. Is there a good reason not to use `.chars()` though?

Reply:
Just that I go through the discussions top to bottom on commenting then a second time on testing :P

Reply:
Now there is a good reason:
```
collapse/dtrace         time:   [12.756 ms 12.946 ms 13.172 ms]
                        thrpt:  [100.09 MiB/s 101.84 MiB/s 103.36 MiB/s]
                 change:
                        time:   [+15.040% +17.531% +19.861%] (p = 0.00 < 0.05)
                        thrpt:  [-16.570% -14.916% -13.074%]
                        Performance has regressed.
Found 6 outliers among 100 measurements (6.00%)
  6 (6.00%) high mild
``` 
:(

Reply:
Ouch, that's pretty severe. And surprising given that it's supposed to be an ASCII string, so there shouldn't even be any multi-byte characters to decode. I guess one thing we could do as a sanity check is also add an arm for each of the [leading code points](https://en.wikipedia.org/wiki/UTF-8#Description) of the multi-byte UTF-8 sequences. That way we'll at least detect if things are about to go terribly wrong.

Reply:
Okay so here comes the research results: dtrace stack parts are c-strings

Looking at [the function responsible for printing the stacktracek](https://github.com/opendtrace/opendtrace/blob/1a03ea5576a9219a43f28b4f159ff8a4b1f9a9fd/lib/libdtrace/common/dt_consume.c#L1385) all variables are `char *` not wide chars so it's c-strings and should be save to be addressed as bytes.

I added the link and an explanation to the code

Reply:
Perfect! Thank you!